pkg/cosign/verify.go

```diff
@@ -725,15 +725,20 @@ func keyBytes(sig oci.Signature, co *CheckOpts) ([]byte, error) {
 	if err != nil {
 		return nil, err
 	}
-	// We have a public key.
-	if co.SigVerifier != nil {
+	switch {
+	// certificate provided from the OCI image, or via flag (--certificate or --bundle)
```

Comment:
is this comment accurate / misleading? If it's OCI it will always take it in from the OCI image, NOT from the `--certificate` flag. I think @nsmith5 issue shows a "Red herring" like you mentioned. I think the issue should have happened if the `--certificate` was specified or not since it's comparing `sig.Cert()` form the OCI annotation. The OCI verify CLI never sets that. So maybe

```	// certificate provided from the OCI image, or, in the case of a blob, via flag (--certificate or --bundle)```